emote/memory/fifo_strategy.py

```diff
@@ -43,6 +48,20 @@ def post_import(self):
                 # as that'd make it very hard to uphold variants over time.
                 self.track(-abs(id) - 1, length)
 
+    def state(self) -> dict:
+        """Serialize the strategy to a JSON-serializable dictionary"""
+        return {
+            "identities": list(self._identities),
+            "sequence_lengths": list(self._sequence_lengths.items()),
+        }
+
+    def load_state(self, state: dict):
+        """Load the strategy from a dictionary"""
+        self._identities = deque(state["identities"])
+        self._sequence_lengths = dict(state["sequence_lengths"])
+
+        self._in_simple_import = True
```

Comment:
This is confusing to me. Where is this unset? I can't figure out this logic. It feels like this flag should be handled by `begin_simple_import` and `end_simple_import`. Why does it need to be handled explicitly here? And why does that happen at the end of load_state?

Reply:
It indeed should not! Originally `in_simple_import` was managed by each strategy, not through the base class.